### PR TITLE
Run build workflow on change

### PR DESCRIPTION
**Pull request by austinvazquez**
**Issue #, if available:**
Partial #911 

**Description of changes:**
Runs build workflow on changes.

**Testing performed:**
Build workflow ran on this PR.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ on:
   pull_request:
     branches: ['main', 'release/**']
     paths:
+      - '.github/workflows/build.yml'
```
Comment (Kern--) [inline .github/workflows/build.yml:13]: we should include `prebuild.yml` as well.

Comment (sondavidb) [reply .github/workflows/build.yml:13]: Wouldn't doing this also make it so changes in `prebuild.yml` would make `build.yml` also run?

Comment (Kern--) [reply .github/workflows/build.yml:13]: Oh, right. I guess I was thinking prebuild should rerun if prebuild changes, but that's not what this is. Whoops.

**Review by Kern-- — APPROVED** [review]

**Review by sondavidb — APPROVED** [review]
LGTM, we'll need to be a bit more thorough to fully fix #911 but this is a simple quick fix for now.

Thanks for this. I'll merge it immediately so that I can get #1027 to auto-build.

**Pull request merged**